src/core/map/hooks/useMapEffect.ts

```
@@ -0,0 +1,12 @@
+import { useEffect, type DependencyList } from 'react';
+import type { IMap } from '../providers/IMapProvider';
+
+export function useMapEffect<TMap extends IMap>(
+  map: TMap,
+  effect: (map: TMap) => void | (() => void),
```

Biome (1.9.4)

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)